diff -up ./esc/rpm/esc.desktop.fix19 ./esc/rpm/esc.desktop
--- ./esc/rpm/esc.desktop.fix19	2013-07-30 17:33:21.786650959 -0700
+++ ./esc/rpm/esc.desktop	2013-07-30 17:33:32.628653662 -0700
@@ -2,6 +2,6 @@
 Type=Application
 Name=Smart Card Manager
 Comment=Enterprise Security Client Smart Card Manager
-Categories=System;Application;X-Red-Hat-Base;
+Categories=System;X-Red-Hat-Base;
 Exec=esc
 Icon=esc-png
